### PR TITLE
Adds support for custom sale-on-hold banners

### PR DESCRIPTION
**Pull request by ashfurrow**
This is work on https://github.com/artsy/auctions/issues/665 .

@erikdstock the technical changes are quite small. We turn an Observable from just sending `Bool`s to sending a tuple of `(isOnHold: Bool, message: String?)` (the message is optional). If the message exists and is not the empty string, we display it. This is a WIP because I need to update some tests.

@katarinabatina Here's what it looks like on phone:

simulator screen shot oct 16 2017 5 31 43 pm

And on iPad:

simulator screen shot oct 16 2017 5 38 19 pm

Basically, if there is a custom message sent from causality, we display it. We also display "You can still place max bids" under it. Users will see this overlay when the sale goes on hold, if they enter a sale that is already on hold, or if they tap the "Sale on Hold" banner on the main LAI interface:

simulator screen shot oct 16 2017 5 40 36 pm

I'm happy to chat about any updates you want to the design, just let me know 👍

### CODE REVIEW

**Comment by acjay** [comment]
Didn't we want people to still be able to max bid when the sale is on hold?

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">No CHANGELOG changes made</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Needs testing on a Phone if change is non-trivial</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_travis">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by acjay** [comment]
Silly me! I must not have scrolled all the way down past the massive screenshot 😊
